query/src/lib.rs

```diff
@@ -129,6 +150,19 @@ pub enum FnArg {
     Constant(NonIntegerConstant),
 }
 
+impl FromValue<FnArg> for FnArg {
+    fn from_value(v: &edn::Value) -> Option<FnArg> {
+        // TODO: support SrcVars.
+        Variable::from_value(v)
```

Comment:
Is this better than matching the `edn::Value` first?  You don't want to encode knowledge of variables into `FnArg`?